Client/Nimbus/OnboardingFeatureLayer/NimbusOnboardingFeatureLayer.swift

```diff
@@ -0,0 +1,104 @@
+        from cardData: [NimbusOnboardingCardData],
+        using cardOrder: [String]
+    ) -> [OnboardingCardInfoModel] {
+        let cards = getOnboardingCards(from: cardData)
+
+        // Sorting the cards this way, instead of a simple sort, to account for human
+        // error in the order naming. If a card name is misspelled, it will be ignored
+        // and not included in the list of cards.
+        return cardOrder.compactMap { cardName in
+            guard let card = cards.first(where: { $0.name == cardName }) else { return nil }
+            return card
+        }
+    }
+
+    /// Converts ``NimbusOnboardingCardData`` to ``OnboardingCardInfoModel``
+    /// to be used in the onboarding process.
+    ///
+    /// All cards must have valid formats and data. For example, a card with no
+    /// buttons, will be omitted from the returned cards.
+    ///
+    /// - Parameter cardData: Card data from ``FxNimbus/shared``
+    /// - Returns: An array of viable ``OnboardingCardInfoModel``
+    private func getOnboardingCards(from cardData: [NimbusOnboardingCardData]) -> [OnboardingCardInfoModel] {
+        return cardData.compactMap { card in
+            let image = getOnboardingImageID(from: card.image)
+            guard let buttons = getOnboardingCardButtons(from: card.buttons),
+                  !buttons.isEmpty
+            else { return nil }
+
+            return OnboardingCardInfoModel(name: card.name,
+                                           title: card.title,
+                                           body: card.body,
+                                           image: image,
+                                           link: getOnboardingLink(from: card.link),
+                                           buttons: buttons,
+                                           type: card.type)
+        }
+    }
+
+    /// Returns an optional array of ``OnboardingButtonInfoModel`` given the data.
+    /// A card is not viable without buttons.
+    private func getOnboardingCardButtons(from cardButtons: [NimbusOnboardingButton]) -> [OnboardingButtonInfoModel]? {
+        if cardButtons.isEmpty { return nil }
+
+        return cardButtons.map { OnboardingButtonInfoModel(title: $0.title, action: $0.action) }
+    }
+
+    /// Returns an optional ``OnboardingLinkInfoModel``, if one is provided. This will be
+    /// used by the application in the privacy policy link.
+    private func getOnboardingLink(from cardLink: NimbusOnboardingLink?) -> OnboardingLinkInfoModel? {
+        guard let cardLink = cardLink,
+              let url = URL(string: cardLink.url)
+        else { return nil }
+
+        return OnboardingLinkInfoModel(title: cardLink.title, url: url)
+    }
+
+    /// Translates a nimbus image ID for onboarding to a an ``ImageIdentifiers`` based id
+    /// that corresponds to an app resource.
+    ///
+    /// In the case that a unknown image identifier in entered into experimenter, the
```

Comment:
nit, I think that's what you meant?
```suggestion
    /// In the case that an unknown image identifier is entered into experimenter, the
```